feat: Add comprehensive Fabric monitoring and logging solution (801-m…

shashax42 opened the pull request:
##  Summary
Request to add a comprehensive monitoring and logging solution example for Microsoft Fabric resources to enhance operational visibility and alerting capabilities.

##  Motivation
Currently, the quickstart examples focus on resource provisioning but lack operational monitoring examples. This creates a gap for users who need to:
- Monitor Fabric Capacity performance and utilization
- Set up intelligent alerting for resource thresholds
- Implement centralized logging for troubleshooting
- Create real-time dashboards for operational insights

## 💡 Proposed Solution
Add a new quickstart example `801-monitoring-logging` that includes:

### Core Components
- **Log Analytics Workspace**: Centralized log collection and analysis
- **Application Insights**: Application performance monitoring
- **Azure Dashboard**: Real-time metrics visualization
- **Alert Rules**: Intelligent alerting for CPU, memory, and storage thresholds

### Key Features
- ✅ Configurable alert thresholds
- ✅ Email and webhook notifications
- ✅ Multi-environment support (dev/staging/prod)
- ✅ Cost optimization features
- ✅ Comprehensive testing (unit + acceptance)
- ✅ Complete documentation with troubleshooting guide

##  Implementation Details
- Follow existing project patterns and conventions
- Include Terraform tests using TDD methodology
- Provide example configurations and best practices
- Support resource tagging and naming conventions

##  Expected Benefits
- Enhanced operational visibility for Fabric deployments
- Proactive issue detection and alerting
- Reduced MTTR (Mean Time To Recovery)
- Better resource utilization insights
- Improved troubleshooting capabilities

##  Ready for Implementation
I have already implemented this solution and it's ready for review in the `feature/operations-management` branch.

shashax42 commented:
> @shashax42입력한 명령이 잘못되었습니다. 다시 시도해 주세요.
> 
> 예를 들면 다음과 같습니다.
> 
> ```
> @microsoft-github-policy-service agree
> ```
> 
> 그리고
> 
> ```
> @microsoft-github-policy-service agree company="your company"
> ```

@microsoft-github-policy-service agree
 @microsoft-github-policy-service agree company="Personal"

DariuszPorowski commented:
@badeamarjieh @OrBaubergMicrosoft @Halamish51 PTAL for this PR, looks very interesting!

@shashax42 please rename 801 to 402, we keep 100-400 scale for complexity level indicator, this example fits into 400 level, and the next available number in-line is 2 :) thanks!

shashax42 commented:
@DariuszPorowski 
Thank you for the feedback🙏I followed your suggestion and would appreciate it if you could take a look. Thank you✨

badeamarjieh commented on quickstarts/402-monitoring-logging/main.tf:
```diff
@@ -0,0 +1,193 @@
+# Diagnostic Settings for Fabric Capacity
+resource "azurerm_monitor_diagnostic_setting" "fabric_capacity_diagnostics" {
+  name                       = "diag-fabric-capacity-${var.solution_name}"
+  target_resource_id         = data.fabric_capacity.monitored_capacity.id
```
the capacity id from fabric resource is different from the capacity id in azure.
in fabric, the capacity id is a GUID, whereas in azure its just like any id including the URL identifier.
is this the intended use?

badeamarjieh commented:
Thanks @shashax42 for the elaborate explanation!

@Halamish51 PTAL. Code-wise it's approved from my end.

shashax42 commented:
@badeamarjieh Updated the big titles to “Fabric Capacity Monitoring …” to clearly distinguish it from workspace monitoring in the UI. Thanks for the suggestion!
Follow-up: adjusted the 402 example for azurerm v4.x compatibility (portal dashboard + Fabric Capacity ARM ID) and updated diagnostic settings to address deprecation warnings. `terraform validate` passes locally.

shashax42 commented:
@badeamarjieh 
Thank you so much for your kindness and for being so proactive and helpful with my contributions!
